# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCaret.h
# Commit: a69c40accbf0
# Full Hash: a69c40accbf099d041417ce4da9b82829f044167
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2023-12-06 21:30:12
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a bug where caret misses when dragging and dropping r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192377
# ==============================================================================

diff -r c9558e7c5c2d -r a69c40accbf0 layout/base/nsCaret.h
--- a/layout/base/nsCaret.h	Wed Dec 06 16:17:57 2023 +0000
+++ b/layout/base/nsCaret.h	Wed Dec 06 16:29:18 2023 +0000
@@ -13,6 +13,7 @@
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/dom/Selection.h"
 #include "nsCoord.h"
+#include "nsIFrame.h"
 #include "nsISelectionListener.h"
 #include "nsIWeakReferenceUtils.h"
 #include "CaretAssociationHint.h"
@@ -284,6 +285,11 @@
    * it's in non-user-modifiable content.
    */
   bool mIgnoreUserModify;
+
+  /**
+   * mLastCaretFrame is the last caret frame that was scheduled to paint.
+   */
+  WeakFrame mLastCaretFrame;
 };
 
 #endif  // nsCaret_h__